examples/java-matter-controller/java/src/com/matter/controller/commands/common/RealResult.java

```diff
@@ -0,0 +1,51 @@
+/*
+ *   Copyright (c) 2023 Project CHIP Authors
+ *   All rights reserved.
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License");
+ *   you may not use this file except in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ *
+ */
+
+// RealResult encapsulate the boolean result, and error string if the result is false.
+package com.matter.controller.commands.common;
+
+public class RealResult implements Result {
+  private boolean result;
+  private String error;
+
+  public RealResult() {
+    this.result = true;
+  }
+
+  public RealResult(String error) {
+    this.result = false;
+    StringBuffer buffer = new StringBuffer(error);
+    this.error = buffer.toString();
+  }
+
+  public static RealResult Success() {
+    return new RealResult();
+  }
+
+  public static RealResult Error(String error) {
+    return new RealResult(error);
+  }
+
+  public boolean getResult() {
```

Comment:
is `@Override` supported to tag this as an override?

Reply:
updated, thanks